src/app/lists/EditModal.tsx

```diff
@@ -0,0 +1,125 @@
+"use client";
+
+import Modal from "@/components/Modal/Modal";
+import React, { useState } from "react";
+import styled from "styled-components";
+import { SnackBarDiv, headers, tooltips } from "@/app/lists/ListDataview";
+import TextInput from "@/components/DataInput/FreeFormTextInput";
+import { Datum } from "@/components/Tables/Table";
+import supabase from "@/supabase";
+import Snackbar from "@mui/material/Snackbar/Snackbar";
+import Alert from "@mui/material/Alert/Alert";
+import Button from "@mui/material/Button/Button";
+
+interface Props {
+    onClose: () => void;
+    /** null => add, undefined => modal closed, Datum => edit */
+    data: EditModalState;
+}
+
+// null => add, undefined => modal closed
+export type EditModalState = Datum | null | undefined;
+
+type HeadersAndTooltips = [[string, string], [string, string]][];
+
+const ModalInner = styled.div`
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+`;
+
+const DataWithTooltipDiv = styled.div`
+    display: flex;
+    gap: 1rem;
+    flex-direction: column;
+
+    @media (min-width: 600px) {
+        flex-direction: row;
+    }
+`;
+
+const DisplayContents = styled.div`
+    display: contents;
+`;
+
+const EditModal: React.FC<Props> = ({ data, onClose }) => {
+    const [toSubmit, setToSubmit] = useState<{ [key: string]: string | null }>(
+        data ? { ...data.data, ...data.tooltips } : {}
+    );
+
+    const [errorMsg, setErrorMsg] = useState<string | null>(null);
+
+    const headersThatHaveTooltips = headers.filter(([key]) => key !== "item_name");
+
+    const headersAndTooltips: HeadersAndTooltips = headersThatHaveTooltips.map((header, index) => {
+        return [header, tooltips[index]];
```

Comment:
FYI you don't _need_ a `{ return ... }` - if you use single brackets then you can just have a value - e.g. `.map((h,i) => [h, t[i]]);` should be sufficient